### ISSUE TITLE
Discussions here

### ISSUE THREAD

**Issue by wookiefriseur** [manavortex/FurnitureCatalogue#59]
We can use this issue if we want to discuss something. That way we don't have to use PRs for that.

**Comment by wookiefriseur** [comment]
@BerylBones Were you planning on doing luxury furnishings this weekend? I wanted to have a look at the PTS but I could also do the luxury furnishings while I'm at it. Either way is fine for me. Let me know

**Comment by BerylBones** [comment]
@wookiefriseur Sorry, busy weekend!

I see you pulled in all the new items as well, thanks for doing that! I'll try to add known sourcing for the new things within a week or so of update 39 launching. Most of them I think are crown items so should be a fairly quick update.

**Comment by wookiefriseur** [comment]
Nice, I was not sure how to best get the sources.

Added some stuff and prepared for the next update but I'll wait with the next release until the patch is out.

In the meantime I'll keep trying to fix issues and maybe have a jab at search/load performance.

**Comment by wookiefriseur** [comment]
@BerylBones I hope my recent changes didn't make it harder for you to contribute. Let me know if you need help with anything. I'm thinking about making a webinterface for the database, so it'll become easier to maintain.

**Comment by BerylBones** [comment]
@wookiefriseur I am so sorry that I disappeared on you. I've had one crisis after another and I should have reached out to let you know I'd be gone for a while.

But, I am back now and ready to get this updated. I've pulled your latest update from the dev branch and will be working on it today. Let me know if anything else has changed!

**Comment by wookiefriseur** [comment]
No worries, good to have you back. 

I did not notice many things missing ingame so it was low priority for me too. Mostly just did luxury furnishing updates and some missing items. But it's perfect timing, now that the new update just dropped.

So far I only prepared version updates for it and the grandmaster crafting stations. Will push it in a minute.

**Comment by BerylBones** [comment]
I'm almost ready to push this update, but I'm not sure of the best way to handle the new achievement furnishings. They are purchased with archival fortunes instead of gold. For now I can use the existing setup, but let me know if you think of a way to note the different currency.

**Comment by manavortex** [comment]
Back in the day, I handled that by shoving it into a different table (for the IC currency). By now, it might be easier to give them a currency flag?

**Comment by wookiefriseur** [comment]
Yeah, will probably use the simple text for now, because currently there are only a handful of items. 

But it seems like there might be a weekly rotation for one of the Endless Archive traders, so maybe I'll make a separate file later (if it is similar to the luxury furnishings)

**Comment by BerylBones** [comment]
The merchant with the rotating inventory shouldn't have anything that's exclusive to him. Everything he sells you can get as a drop somewhere, so I wouldn't worry about setting anything up for him.

**Comment by manavortex** [comment]
In that case, why not have a list with only item IDs that this guy can 
possibly say, and then concat `or sometimes sold by <this one dude>` to 
the tooltip if it’s in there?

**Comment by BerylBones** [comment]
Yes, we can definitely do that. I did not for this update, but I've made a note of which items he has this week so I can add it in for next time.

**Comment by BerylBones** [comment]
I am apparently *very* behind on updates, so thank you so much @wookiefriseur for keeping this going. I didn't realize how far behind I was until I tried to push Gold Road items today and barely recognize the addon lol. I think I got it mostly figured out (minus the actual publishing), but please let me know if I messed anything up. I'm hoping to dig into the rumor item sources this week while I have some extra time.

**Comment by BerylBones** [comment]
I'm a quick study! 😁 When you have time, a walkthrough the publishing process would be great! No rush through, and I'll work on item sourcing in the meantime!

**Comment by wookiefriseur** [comment]
Alright. I updated [CONTRIBUTING](https://github.com/manavortex/FurnitureCatalogue/blob/master/docs/CONTRIBUTING.md#recommended-release-flow). You can see the general flow in the diagram, maybe it helps. 

I should have time to walk you through a release later this week. Won't take long. More than enough potential release dates to pick from. And there is a luxury furnishing update every week, so even that would work as a release 😃 (that smiley looks a bit unhinged, when you zoom all the way in)

**Comment by wookiefriseur** [comment]
The `dev` branch is now up to date with the `master` and on on top of it are your recipe-PR #141  and my string overhaul PR #142 

I wasn't able to test the English client yet, so maybe some strings still look strange there. But all in all this is ready for release. You can check the dev branch out and do your item sourcing on that version (if you haven't already started).

Let me know if you run into any problems

**Comment by wookiefriseur** [comment]
Ohh, @BerylBones  I just remembered that you probably don't have merge permission on this repo. We used to work on your fork, then you made pull requests, then manavortex merged it. I only got full merge permission, because I asked to be maintainer.

So there is not much to walk you through right now regarding publishing, unless you have permission or get permission. But nevertheless just let me know ingame or on Gitter/Element/Matrix if you need help with anything, for instance with `git` or the code. I can also share my screen.

----

**The whole process for you right now is like this:**

1. pull the latest `dev` branch from the main repo (here)
2. make some changes
3. pull from time to time, in case `dev` on the main repo has changed
   - `git pull https://github.com/manavortex/FurnitureCatalogue.git dev`
   - or `git pull --rebase https://github.com/manavortex/FurnitureCatalogue.git dev`
4. when you're done make a PR to `dev` on the main repo (you can just delete the prefilled PR-text and make your own)

----

**After that comes my part:**

1. I merge your PR with the other stuff
2. then I make a new PR from `dev` to `master`, both on the main repo
3. I copy any CHANGELOG relevant notes from your PR
4. I set labels for the release, so the bot knows what to do
5. then I merge on `master`

----

**And now the automated parts:**

1. the bot is triggered, updates the AddOn version and generates some files for us
2. the bot creates a PR from the branch `release/1.2.3` to `master` with those changes (like #131 )
3. I check if everything looks fine and then I merge it
4. this triggers a different bot action and it creates a release [here](https://github.com/manavortex/FurnitureCatalogue/releases)
5. the bot then publishes the release on ESOUI

**Comment by manavortex** [comment]
@BerylBones added you as collaborator :)

**Comment by BerylBones** [comment]
Perfect, thank you both! I have a couple more item sources I want to add in today, and then we'll give this a shot!

**Comment by BerylBones** [comment]
Ok, I think I missed a step somewhere. I managed to get everything merged onto the master (set label as 'release'), but it doesn't look like it triggered the automated process.

**Comment by wookiefriseur** [comment]
Oh yeah, I left a comment in #144 . The automatic script tried to format the files before creating the release but then encountered a syntax error. We can either roll it back or do a small patch-PR from `dev` to `master` and try again. 

Other than that the steps looked all good 👍

**Comment by BerylBones** [comment]
Ok, trying it with a small patch-PR

**Comment by BerylBones** [comment]
I think I found it. Missing comma in the vvardenfell painting sourcing

**Comment by wookiefriseur** [comment]
Ok, you know the drill. I hope it works this time. But so far it's good that the script caught those.

**Comment by BerylBones** [comment]
Woo! Third time's the charm!

**Comment by wookiefriseur** [comment]
So glad it worked. My palms were sweaty all the time because I had not tested the new bot script yet. 😁

The release should show up on ESOUI soon.

Good job 🥳

**Comment by BerylBones** [comment]
Looks like everything is working! Thanks for walking me through everything.

**Comment by wookiefriseur** [comment]
I prepared the luxury furnishings for this weekend but hadn't time to add them yet. And it looks like you already added 1 of this weekend's luxury furnishings.

If you want to practice some more, you can also do the rest tomorrow and make another release, this time with a `MINOR` label.

Here, if this helps (in German, but the IDs are there):
```lua
	[134465] = {		-- Varlastein, glühend
		itemPrice = 5000,		-- Gold
	},
	[134466] = {		-- Ayleïdenhalterung, leer
		itemPrice = 4000,		-- Gold
	},
	[134467] = {		-- Culandastein, glühend
		itemPrice = 5000,		-- Gold
	},
	[134468] = {		-- ayleïdischer Schalter, uralt
		itemPrice = 4000,		-- Gold
	},
	[156664] = {		-- Ayleïdensäule, klein und leer
		itemPrice = 5000,		-- Gold
	},
	[171421] = {		-- Ayleïdensäule, groß und leer
		itemPrice = 14000,		-- Gold
	},
	[182629] = {		-- ayleïdische Feuerschale, Stein
		itemPrice = 10000,		-- Gold
	},
	[193801] = {		-- Ayleïden-Trennwand, gewölbt
		itemPrice = 20000,		-- Gold
	},
	[203582] = {		-- ayleïdischer Durchgang, hoch und gewölbt
		itemPrice = 20000,		-- Gold
	},
```

If not, I'll just do it on monday or so

**Comment by BerylBones** [comment]
Oh right, I forgot about changing the date on the older lux items. I can do that, and I need to fix the sourcing for gold road pickpocket items.

**Comment by wookiefriseur** [comment]
Alright, thanks! Don't forget the `MINOR` label. That should update it to `4.82.1`

I also edited the script to actually save all lua files changed by the formatter (that part was missing, woops 😊 ).

Oh, yeah I see the pickpocketing:

```lua
-- generic formatter is like :
strGeneric(PREFIX, ADDITIONALINFO, SOURCTYPE, SOURCE)

 -- that should do it for pickpocket in Westworld in general:
local pickpocket_weald = strGeneric(srcPick, nil, "loc", loc.WEALD)

-- if it's from any specific npc types in Westworld, you could do:
local nob_thieves = strSrc("other", npc.CLASS_NOBLE, npc.CLASS_THIEF)
local pickpocket_weald_nob_thieves = strGeneric(srcPick, nob_thieves, "loc", loc.WEALD)
```

**Comment by wookiefriseur** [comment]
Well done. 1st try. I should have said `PATCH` instead of `MINOR`. Got it mixed up, but not important 😄

**Comment by BerylBones** [comment]
@wookiefriseur did you contribute at all to ESO-Hub's new furniture catalogue tool? I've already received enough questions about it that I'm about to email them to ask them to change its name, but I don't want to do that if you're part of the project. 🙃

**Comment by wookiefriseur** [comment]
@BerylBones  No, first time I'm hearing of this. I wouldn't do anything major without talking about it first. 

Looks like what MMO Fashion used to do, but for furniture. Yeah, that name can cause some confusion. I'm currently sick so I can't think up a good alternative name. 🤧

**Comment by wookiefriseur** [comment]
Sure, looks like Fallen Banners version might have been removed and maybe some crown crate name. Nothing tragic.

Easiest way might be to compare the current dev with master or just send the PR and then look at the difference 😁 

I'll have another look.

Does it work ingame? I cannot test it at the moment.

**Comment by BerylBones** [comment]
Yes, it does work in game.

Ok, that's not too bad then, thank you! I was updating it while distracted and saw changes where I wasn't expecting changes so I panicked a little 😅

**Comment by wookiefriseur** [comment]
You Thanos-snapped the previous april release out of existence but no biggie (JP translations and some other stuff). 

If you want to play around with git you could fix it by backing up your changes somewhere, then doing a hard reset:

```bash
# while being in your local dev branch
git reset --hard master
```

And then paste your changes back in.

But no worries: If it doesn't break ingame you can release it and I will fix it later.

> Ok, that's not too bad then, thank you! I was updating it while distracted and saw changes where I wasn't expecting changes so I panicked a little 😅

Next time fetch and pull first before working on the code again and it should be fine 😸

**Comment by BerylBones** [comment]
I have all of my changes backed up so I'll mess around with it. Thanks!

**Comment by BerylBones** [comment]
Ok, that probably should've been a minor release instead of major, but I think I fixed all the things I messed up and will be more careful next time. Thanks again!

**Comment by wookiefriseur** [comment]
Nice, good job fixing it. Sorry I wasn't much help, been really busy the past few days 😵‍💫

**Comment by wolfstar07** [comment]
@wookiefriseur I've updated all relevant files in the depo. I just plan on updating sourcing for more rumor items (I've datamined U50 furnishings) and keep on top of the lux items for the next few weeks. I will also have to add Hearts Week, but other than that, I don't think there are any new systems that need to be added. I'd integrated Tamriel Tomes, Gold Coast Bazaar, and Night Market vendors and am unaware of anything else being added for U50. What would make sense for you to work? I don't know how much the database work would clash with me updating sourcing.

**Comment by wookiefriseur** [comment]
Ah, then you already answered my question in #41 . Nice, you've been busy.

Then I will try to stay clear of the db source files. Should be conflict free. I will probably add API functions first and then slowly replace the calls to point them to the API instead of directly querying DB. That should not cause any merge conflicts.

**Comment by wolfstar07** [comment]
Alright, sounds great. I don't know how long your work will take, but I wasn't anticipating publishing another update until sometime in June (hadn't even originally planned to release today, but it was a lot of work done). I was really trying to get decorators and crafters updated info because I know how frustrating it is to waste crowns on something I didn't have to, but I'll probably take a break this weekend and actually play the game. :P

But once all the sourcing is complete, and until we have the new library, I'll probably look at updating monthly just for the lux. Jhart maintains the lux list on his website (although I discovered he's missing a few items), so I think that's good enough. Users don't need up-to-the-minute accurate info, as long as the updates are in a reasonable time frame.
